odo/numpy_dtype.py

```diff
@@ -23,6 +25,9 @@ def unit_to_dtype(ds):
     if isinstance(ds, DataShape):
         ds = ds.measure
     if isinstance(ds, Option) and isscalar(ds) and isnumeric(ds):
+        with ignoring(NameError):
+            if isinstance(ds.ty, Decimal):
+                return unit_to_dtype(str(ds.ty.to_numpy_dtype()).replace('int', 'float'))
```

Comment:
OK, it doesn't return an error as-is. What `unit_to_dtype` does for an Option(Decimal(9, 0)) is this:

1. `if isinstance(ds, Option) and isscalar(ds) and isnumeric(ds):` will be True
2. It evaluates `str(ds).replace('int', 'float').replace('?', '')`, which gives `decimal[9, 0]` (it strips off the '?')
3. It calls `unit_to_dtype('decimal[9, 0]')`, which returns `dtype('int32')`.

What we'd expect, since this is an Option int, is for it to return a `dtype('float32')`. This doesn't happen because the replacement of `int` with `float` doesn't take place since the Decimal isn't revealed as an `int` until after the replace calls. So this change does the replace on `str(ds.ty.to_numpy_dtype())` instead for Option(Decimal) types. This preserves the behavior that Option ints will be floats in pandas Dataframes.